config/environment.go

```diff
@@ -26,14 +26,36 @@ func GetEnvironment() map[string]string {
 }
 
 func applyEnvKey(key, value string, rValueSubject reflect.Value) {
+	var rFieldValue reflect.Value
 	keyParts := strings.SplitN(key, "_", 2)
 	if len(keyParts) < 1 {
 		return
 	}
-	rFieldValue := rValueSubject.FieldByNameFunc(func(candidate string) bool {
-		candidateUpper := strings.ToUpper(candidate)
-		return candidateUpper == keyParts[0]
-	})
+	if rValueSubject.Kind() == reflect.Struct {
+		rFieldValue = rValueSubject.FieldByNameFunc(func(candidate string) bool {
+			candidateUpper := strings.ToUpper(candidate)
+			return candidateUpper == keyParts[0]
+		})
+	} else if rValueSubject.Kind() == reflect.Map {
+		keys := rValueSubject.MapKeys()
+		for _, k := range keys {
+			if strings.ToUpper(k.String()) == keyParts[0] {
+				el := rValueSubject.MapIndex(k)
+				if el.Kind() == reflect.Map {
+					rFieldValue = reflect.MakeMap(el.Type())
+					for _, sk := range el.MapKeys() {
+						e := reflect.New(el.MapIndex(sk).Type()).Elem()
+						e.Set(el.MapIndex(sk))
+						rFieldValue.SetMapIndex(sk, e)
+					}
+				} else {
+					rFieldValue = reflect.New(el.Type()).Elem()
+					processRFieldValue(rFieldValue, keyParts, value)
+				}
+				rValueSubject.SetMapIndex(k, rFieldValue)
+			}
```

Comment:
Does this overwrite any existing config values? This would include config settings for this plugin and any other plugin.

Reply:
Yes, this function overrides map on each step, but it will not rewrite existing values without reason i.e. no environment variables are set. It also saves everything that was set before. 

Basically works like it was previously with entire config

Reply:
sure, i'll do that next week.
i have some workload issue now